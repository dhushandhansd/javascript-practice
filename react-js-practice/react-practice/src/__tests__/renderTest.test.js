import React from "react";
import { create, act } from "react-test-renderer";
import ButtonEle from "../testing/ButtonEle";

const nameButton = create(<ButtonEle label="Shree Dhushandhan"/>)

test("Button Render with Snapshot Testing", () => {
  expect(nameButton).toMatchSnapshot()
})

test("Button Instances Testing", () => {
  const {root} = nameButton

  const buttonInstance = root.findByType("button")
  const buttonInstanceById= root.findByProps("")

  expect(buttonInstance).toBeTruthy()
  expect(buttonInstanceById).toBeTruthy()
})

test("Button Press Testing", () => {
  const buttonInstance = nameButton.root.findByProps({testId : 'name-button'}).props
  act(() => buttonInstance.onClick())

  const textInstance = nameButton.root.findByProps({testId : 'name-label'}).props
  expect(textInstance.children).toEqual('button pressed')

})