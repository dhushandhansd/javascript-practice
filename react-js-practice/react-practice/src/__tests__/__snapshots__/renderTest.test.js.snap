// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render with Snapshot Testing 1`] = `
<div>
  <button
    data-testId="name-button"
    onClick={[Function]}
  >
    Shree Dhushandhan
  </button>
</div>
`;
